core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -175,6 +175,13 @@ class Partition(val topic: String,
       // add replicas that are new
       allReplicas.foreach(replica => getOrCreateReplica(replica))
       val newInSyncReplicas = partitionStateInfo.isr.asScala.map(r => getOrCreateReplica(r)).toSet
+
+      // If the ISR size received from a LeaderAndIsrRequest is smaller than the existing ISR size,
+      // mark the isrShrinkRate meter. For example: This can happen if one of the followers is shutdown.
+      // Note that this metric is not triggered on follower -> leader transition
```

Comment:
Follow-up to Jun's question above on this comment? I think what Aditya is trying to convey here is that the `isrShrinkRate` can only record shrinks for the duration of leadership. So for e.g., there is no cluster-wide metric to capture shrinks caused by (say) a leader's hard-failure and the controller subsequently electing a new leader (with a shrunk ISR). Perhaps it is better to remove this comment and instead add comments to the place where `isrShrinkRate` meter is defined and describe there under what circumstances the shrink rate is and is not updated.